src/fluree/db/flake/index/novelty.cljc

```diff
@@ -321,6 +321,54 @@
                      "Error writing novel index node:" display-node)
           (async/>! error-ch e))))))
 
+(defn- compute-stats-from-novelty
+  "Core logic for computing property and class counts from novelty flakes.
+   Increments for assertions (op=true), decrements for retracts (op=false)."
+  [novelty-flakes prev-properties prev-classes]
+  (loop [[f & r] novelty-flakes
+         properties prev-properties
+         classes prev-classes]
+    (if f
+      (let [p     (flake/p f)
+            delta (if (flake/op f) 1 -1)
+            properties* (update-in properties [p :count] (fnil + 0) delta)
+            classes*    (if (flake/class-flake? f)
+                          (update-in classes [(flake/o f) :count] (fnil + 0) delta)
+                          classes)]
+        (recur r properties* classes*))
+      {:properties properties
+       :classes    classes})))
+
+(defn compute-novelty-stats
+  "Computes property and class counts from novelty flakes in a separate thread/go block.
+   Returns a channel that will contain the computed statistics.
+
+   Uses async/thread on JVM for true parallelism (not limited by go block thread pool).
+   Falls back to go block on ClojureScript."
+  [novelty-flakes prev-properties prev-classes]
+  #?(:clj
+     (async/thread
+       (compute-stats-from-novelty novelty-flakes prev-properties prev-classes))
+     :cljs
+     (go
+       (compute-stats-from-novelty novelty-flakes prev-properties prev-classes))))
+
+(defn current-stats
+  "Compute current property and class statistics."
+  [db]
+  (let [indexed-stats     (get db :stats {})
```

Comment:
`db` will always be a FlakeDB returned by the `branch/current-db` fn, so it will always have the `:stats` key - the fallback empty map isn't necessary.